egi_notebooks_hub/d4science.py

```diff
@@ -361,3 +367,15 @@ async def pre_spawn_hook(self, spawner):
                     },
                 }
             ]
+            if self.ophidia_image != "":
+                ophidia = {
+                    "name": "ophidia",
+                    "image": self.ophidia_image,
+                    "volumeMounts": [
+                        {
+                            "mountPath": "/home/jovyan/dataspace",
+                            "name": "dataspace-shared",
```

Comment:
I feel this could be better added as an extra configuration in python to the specific deployment, no?

Reply:
Something like:
```
class OphidiaSpawner(D4ScienceSpawner):
    async def pre_spawn_hook(self, spawner):
        await super().pre_spawn_hook(spawner)
        ophidia = {
            "name": "ophidia",
            "image": "whatever",
            "volumeMounts": [
                {
                    "mountPath": "/home/jovyan/dataspace",
                    "name": "dataspace-shared",
                },
            ],
        }
        spawner.extra_containers.append(ophidia)
```

Reply:
The thing I don't like is the hardcoded paths, but maybe I'm just a bit pedantic here.

I'm also thinking on potentially moving away from KubeSpawner in the future (for using more than one provider), but that will require changes all over the place so probably for having better control of what needs to change is better if everything is in the hub code directly, so approving :)